test/__snapshots__/test.js.snap

```diff
@@ -1,5 +1,75 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`with processors AsyncCallbackInterface.webidl 1`] = `
+"\\"use strict\\";
+
+const conversions = require(\\"webidl-conversions\\");
+const utils = require(\\"./utils.js\\");
+
+const interfaceName = \\"AsyncCallbackInterface\\";
+
+exports.convert = function convert(value, { context = \\"The provided value\\" } = {}) {
+  if (!utils.isObject(value)) {
+    throw new TypeError(\`\${context} is not an object.\`);
+  }
+
+  const { wrapper } = {
+    wrapper(...args) {
+      let completion;
+      let thisArg = this;
+      let O = value;
+      let X = O;
+
+      try {
+        if (typeof O !== \\"function\\") {
+          X = O[\\"asyncMethod\\"];
+          if (typeof X !== \\"function\\") {
+            throw new TypeError(\`\${context} does not correctly implement AsyncCallbackInterface.\`);
+          }
+          thisArg = O;
+        }
+
+        for (let i = 0; i < args.length; i++) {
+          args[i] = utils.tryWrapperForImpl(args[i]);
+        }
+        if (args.length < 0) {
+          for (let i = args.length; i < 0; i++) {
+            args[i] = undefined;
+          }
+        } else if (args.length > 0) {
+          args.length = 0;
+        }
+
+        let callResult = Reflect.apply(X, thisArg, args);
+
+        callResult = Promise.resolve(callResult).then(
+          value => {
+            value = conversions[\\"any\\"](value, { context: context + \\" promise value\\" });
+
+            return value;
+          },
+          reason => reason
+        );
+
+        return callResult;
+      } catch (err) {
+        return Promise.reject(err);
+      }
+    }
+  };
+
+  Object.defineProperties(wrapper, {
+    [utils.wrapperSymbol]: { value },
+    value: { value, enumerable: true }
```

Comment:
So the use case for value is for things like https://dom.spec.whatwg.org/#add-an-event-listener's "callback is listener’s callback". However, per https://github.com/whatwg/dom/issues/842 I think this should really be comparing the "object reference".

What do you think of calling this `objectReference` instead, to be closer to the spec? Then, if https://github.com/whatwg/dom/issues/842 gets fixed, it will be more obvious how we're matching it.

Alternately we could advocate for changing the spec's concept name from "object reference" to something nicer (perhaps even "value").

Reply:
If I switch this to use `WeakMap`s, then calling `convert` on the same value multiple times will result in identical wrapper functions.

Reply:
That seems incorrect. Per spec a new converted (not wrapper) IDL value is created each time. We should follow that and not add complex caching that isn't needed.